pkg/api/nextdate.go

```diff
@@ -0,0 +1,226 @@
+package api
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const DateFormat = "20060102"
+
+// NextDate обработка правил вычисления следующей даты.
+func NextDate(now time.Time, dstart string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", fmt.Errorf("empty repeat rule")
+	}
+
+	date, err := time.Parse(DateFormat, dstart)
+
+	if err != nil {
+		return "", fmt.Errorf("invalid date format: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
		return "", fmt.Errorf("invalid date format: %w", err)
```